Make misa.y=0 condition implementation defined (#1144)
riscv/riscv-cheri: src/cheri/riscv-priv-integration.adoc

NOTE: The list includes <<ddc>> if <<section_cheri_hybrid_ext>> is implemented.

NOTE: This restriction ensures M-mode software cannot drop CHERI checks by modifying <<misa_y,misa>> once they have been configured.

>  The exact behavior of the restriction is implementation defined, and can be as simple as detecting whether any CSR writes or <<pcc>> metadata updates have been executed in {cheri_cap_mode_name}.

Author: ```suggestion
 The exact behavior of the restriction is implementation defined, and can be as simple as detecting whether any  metadata updates to the relevant CSRs, or the <<pcc>>, have been executed at all in {cheri_cap_mode_name}.
```

>  The exact behavior of the restriction is implementation defined, and can be as simple as detecting whether any  metadata updates to the relevant CSRs, or the <<pcc>>, have been executed at all in {cheri_cap_mode_name}.

Reviewer: That's not the same thing as in the normative text?

Author: That's true - the point is that it covers the case that the normative text needs, and allows more flexibility.
Realistically, software has either set misa.y=0 immediately on boot or isn't going to at all.
Dynamically checking that the PC, DDC etc. are all infinite is annoying, especially for the PC with multiple versions of the metadata in flight in an OoO machine.

it's to allow for simpler implementations which still meet the basic criteria.

Reviewer: I mean, if you're allowing writable misa, you're already accepting that you have to stop your entire pipeline and flush, there's no sensible way that can be done OoO. And the infinite check is easy, you just check against hard-coded bit patterns.

Author: ```suggestion
 The exact behavior of the restriction is implementation defined, and does not need to, for example, check that capability bounds have been narrowed and then reprogrammed _back_ to infinity. Detecting any bounds reduction is sufficient.
```
